Add mysql file handling to mysql module and state

mchugh19 opened the pull request:
This is an attempt to work around the issues described in https://groups.google.com/forum/#!topic/salt-users/Zc4D7NN5RKQ

In short, many salt states rely on the mysql.create_db state, but then install the mysql client in order to populate the database. This overcomplicates the state handling as the mysql client package name must now be tracked for every state doing this behavior and is unusable in an environment not directly connected to the internet. 

### What does this PR do?
This pull request attempts to solve this issue by extending the mysql module with a file_query function where it attempts to read a sql file and run a query for each statement up until the trailing ; and ignores -- comments. In order to detect errors from the mysql module, it also modifies the standard query behavior to return False on error instead of an empty dict. This will cause all mysql processing to stop on any error, which I would think is the expected behavior. 

It also adds a run_file state allowing for use such as:
```
import_schema:
  mysql_query.run_file:
    - database: some_db
    - query_file: /tmp/schema.sql
```

With resulting output:
```
local:
----------
          ID: import_schema
    Function: mysql_query.run_file
      Result: True
     Comment: {'rows affected': 1L, 'query time': {'raw': 21.2241, 'human': '21.22s'}}
     Started: 17:36:00.136384
    Duration: 23852.153 ms
     Changes:
              ----------
              query:
                  Executed
```


### Tests written?
No

Commit:
Modify mysql module to return False on failure
Add file hanlding to mysql module
Add file hanlding to mysql state

cachedout commented:
@mchugh19 Very nice. There are a few lint errors here that we need to have cleaned up before we can get this in. I would also like @gtmanfred to have a look at this.

gtmanfred commented on salt/modules/mysql.py:
```diff
@@ -676,6 +676,49 @@ def query(database, query, **connection_args):
         return ret
 
 
+def file_query(database, file_name, **connection_args):
```
Need a doc string here to show how to use the module on the commandline

mchugh19 replied:
Done

gtmanfred reviewed (changes requested):
Other than the one change.  The rest of this looks solid.

Thanks!

Commit:
Add versionadded string
Clean up some lint errors
Add usage docstring

mchugh19 commented:
Added docstring, versionadded string, and fixed lint errors.
I think the almost complete duplication of the existing mysql run function is a bit awkward, so if there are any suggestions on re-use, I'm game. But otherwise, I think this likely works.

Commit:
Add missing import of `os`

cachedout commented on salt/modules/mysql.py:
```diff
@@ -676,6 +677,68 @@ def query(database, query, **connection_args):
+            query_result = query(database, query_string, **connection_args)
+            query_string = ""
+
+            if query_result == False:
```
This should be `if query_result is False:`

mchugh19 replied:
done

cachedout commented on salt/states/mysql_query.py:
```diff
@@ -49,6 +49,153 @@ def _get_mysql_error():
+    # The database is present, execute the query
+    query_result = __salt__['mysql.file_query'](database, query_file, **connection_args)
+
+    if query_result == False:
```
Also here: `if query_result is False:`

mchugh19 replied:
done

cachedout reviewed (changes requested):
Two small changes are needed and then this should be all ready to go. Thanks!

Commit:
cleanup comparison

bensbox commented:
Is it possible only to import the sql files, if the database is empty?

mchugh19 commented:
It really just runs all of the sql statements in the file and is as good as your sql. If they require an empty db, then they can fail on a non-empty db. If the sql statements update existing data, then they can do so. 

Basically since it kicks off a bunch of sql, it is not idempotent and is as useful as your sql.
